### PR TITLE
Use cudnn for dropout by default

### PR DESCRIPTION
**Pull request by roywei**
## Description ##
enable cuddn for dropout after https://github.com/apache/incubator-mxnet/pull/13896
This was turned on only in Gluon dropout layers, but not in mx.nd.Dropout.

Test:
After setting changing the flag, ran the sample code in  #13896, was able to get the same speed reported.

```
import mxnet as mx
a = mx.nd.ones((10, 200, 300, 500), ctx=mx.gpu(0))
a.attach_grad()
mx.autograd.set_recording(True)
%timeit mx.nd.Dropout(a, 0.5, mode='always').wait_to_read()

4.24 ms ± 4.35 µs per loop (mean ± std. dev. of 7 runs, 100 loops each
```

### Essentials ###
Please feel free to remove inapplicable items for your PR.

- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
set cudnn_off default as false

### CODE REVIEW

**Comment by wkcn** [comment]
Thanks for your contribution!

unix-gpu CI raises the following exception:
```bash
C++ exception with description "[07:12:07] /work/mxnet/src/resource.cc:429: Check failed: state_space->ctx.dev_id == stream->dev_id (0 vs. -1) The device id of cudnn dropout state space doesn't match that from stream.
```

**Comment by wkcn** [comment]
Thanks for your contribution!
LGTM.
Wait for the finish of CI.

File: tests/cpp/include/test_op.h [hunk 1]
```diff
@@ -72,7 +72,7 @@ struct GPUStreamScope {
     : opContext_(*opContext) {
     CHECK_EQ(opContext_.run_ctx.stream == nullptr, true)
       << "Invalid runtime context stream state";
-    opContext_.run_ctx.stream = mshadow::NewStream<gpu>(true, true);
+    opContext_.run_ctx.stream = mshadow::NewStream<gpu>(true, true, 0);
```
Comment (wkcn) [inline tests/cpp/include/test_op.h]: Sorry for my mistake.
`opContext_.run_ctx.stream = mshadow::NewStream<gpu>(true, true, opContext_.run_ctx.ctx.dev_id);` may be better, although `opContext_.run_ctx.ctx.dev_id=0`

**Comment by anirudhacharya** [comment]
@mxnet-label-bot add [pr-awaiting-review]

**Comment by roywei** [comment]
@wkcn Thanks a lot for pointing out the fix needed on test!
It seems the last CI run was having problem reporting the correct status. (All jobs have passed, but one showing unfinished).

**Comment by roywei** [comment]
seems empty commit did not trigger CI, trying close and reopen

**Comment by wkcn** [comment]
@roywei It seems that the CI has some problems, and all PRs are waiting for status to be reported.

**Comment by roywei** [comment]
@wkcn  yes, and it might be related to https://github.com/apache/incubator-mxnet/issues/11654

**Comment by wkcn** [comment]
@roywei Hi. The problem of CI has been addressed. Could you please retrigger it? Thanks!

**Comment by wkcn** [comment]
Merged. Thanks!

**Pull request merged**